pkg/controller/elasticsearch/driver/upgrade.go

```diff
@@ -296,24 +329,56 @@ func (d *defaultDriver) MaybeEnableShardsAllocation(
 	return results
 }
 
-func (ctx *rollingUpgradeCtx) prepareClusterForNodeRestart(esClient esclient.Client, esState ESState) error {
+func (ctx *rollingUpgradeCtx) readyToDelete(pod corev1.Pod) (bool, error) {
+	if !supportsNodeShutdown(ctx.esClient.Version()) {
+		return true, nil // always OK to restart pre 7.14
```

Comment:
The comment may be confusing as we have decided to support node shutdown from version 7.15.1.
```suggestion
		return true, nil // always OK to restart pre node shutdown support
```